algorithms/greedy.py

```diff
@@ -1,15 +1,107 @@
-from algorithms.wikiracer import WikiRacer
-
-
-class Greedy(WikiRacer):
-    def __init__(self, max_path_length, doc_sim_func):
-        """
-        Parameters
-        ----------
-        max_path_length : int
-        doc_sim_func : function
-            Takes two wikipedia page titles and returns
-            some measure of their similarity.
-        """
-        super().__init__(max_path_length)
-        self.doc_sim = doc_sim_func
+# from algorithms.wikiracer import WikiRacer
+# import algorithms.helper #import get_linked_pages, get_cos_sim
+import requests
+from typing import List
+from sentence_transformers import SentenceTransformer, util
+
+
+class Greedy():
+    def __init__(self):
+        """
+        Parameters
+        ----------
+        max_path_length : int 
+        doc_sim_func : function
+            Takes two wikipedia page titles and returns
+            some measure of their similarity.
+        """
+        #super().__init__(max_path_length)
+        #self.doc_sim = doc_sim_func
+        self.max_path_length = 18 
+        self.model = SentenceTransformer('paraphrase-MiniLM-L6-v2')
+
+
+
+    def set_max_path(self, max_path_length):
+        if max_path_length < 1:
+            raise ValueError("Max path length must be at least 1")
+        self.max_path_length = max_path_length
+
+
+    def find_path(self, start_page: str, dest_page: str):
+        """
+        Returns a list of Wikipedia page titles representing the path.
+        The `start_page` should be included as the first page of the path
+        and the `dest_page` should be included at the end of the path.
+
+        Should raise `FailedPath` exception if it can't find a path for
+        a reason other than an api call failure.
+        """
+        visited = [start_page]
+        count = 0 #or 1
+        # Need to run a while loop that while we have not reached max path length
+        # Or the destination has to been found
+        compare_value = start_page
+        while count < self.max_path_length:
+            print(count)
+            print(visited)
+            # get Linked Pages - we will only once per title
+            links = self.get_linked_pages(compare_value)
+            get_most_similar_value = self.get_most_similar(links, dest_page, visited)
+            visited.append(get_most_similar_value)
+            if get_most_similar_value == dest_page:
+                return visited
+            compare_value = get_most_similar_value
+            count += 1
+            # get_cos_sim - Pomona -> Claremont Colleges -> Pomonaona 
+            # Call that until we get a unqiue vale
+        raise Exception('Failed Path')

```

Comment:
Raise the exceptions that are defined over here:

https://github.com/TiloRC/p-wikiracer/blob/main/algorithms/wikiracer.py#L15